processing/src/test/java/org/apache/druid/query/QueryContextTest.java

```diff
@@ -372,6 +372,20 @@ public void testDefaultEnableQueryDebugging()
     assertTrue(QueryContext.of(ImmutableMap.of(QueryContexts.ENABLE_DEBUG, true)).isDebug());
   }
 
+  @Test
+  public void testDefaultIsDecoupled()
```

Comment:
Perhaps also test what happens when a completely different, random value is set.  Like "bilyBob"

Reply:
update the testcase:
* renamed it to `testIsDecoupled` 
* added check for when its set to `garbage`

I'm not sure what are the gains of having this as a string based setting;
I don't think it will have more variations...
it could be a boolean with the name like `decoupledMode`